docs/walkthrough.md

```diff
@@ -0,0 +1,231 @@
+```
+...
+Inferences/Second vs. Client Average Batch Latency
+Concurrency: 1, throughput: 265.147 infer/sec, latency 3769 usec
+Concurrency: 2, throughput: 890.793 infer/sec, latency 2243 usec
+Concurrency: 3, throughput: 937.036 infer/sec, latency 3199 usec
+Concurrency: 4, throughput: 965.21 infer/sec, latency 4142 usec
+```
+
+6. Run Model Analyzer to find the best configurations for our model
+
+While Model Analyzer comes pre-installed in the SDK container and supports various modes of connecting to a Triton server,
+for simplicity we will use install Model Analyzer in our `server` container to use the `--triton-launch-mode=local` flag.
+To learn more about other methods of connecting Model Analyzer to a running Triton Server, see the `--triton-launch-mode`
+Model Analyzer flag.
+
+```bash
+# Enter server container interactively
+docker exec -ti triton-server bash
+
+# Stop existing tritonserver process if still running
+# because model-analyzer will start its own server
+SERVER_PID=`ps | grep tritonserver | awk '{ printf $1 }'`
+kill ${SERVER_PID}
+
+# Install model analyzer
+pip install --upgrade pip 
+pip install triton-model-analyzer wkhtmltopdf
+
+# Profile the model
+# NOTE: This may take some time, in this example it took ~10 minutes
+model-analyzer profile \
+  --model-repository=/mnt/models \
+  --profile-models=alexnet \
+  --triton-launch-mode=local \
+  --output-model-repository-path=results
+
+# Summarize the profiling results
+model-analyzer analyze --analysis-models=alexnet
+```
+
+Example Model Analyzer output summary:
+
+> In 41 measurements across 6 configurations, `alexnet_config_4` provides the best throughput: 1676 infer/sec.
+>
+> **This is a 67% gain over the default configuration (1002 infer/sec), under the given constraints.**
+
+| Model Config Name | Max Batch Size | Dynamic Batching | Instance Count | p99 Latency (ms) | Throughput (infer/sec) | Max GPU Memory Usage (MB) | Average GPU Utilization (%) |
+|---|---|---|---|---|---|---|---|
+| alexnet_config_4 | 0 | Enabled | 5/GPU | 22.333 | 1675.98 | 6437 | 31.7 |
+| alexnet_config_3 | 0 | Enabled | 4/GPU | 43.26 | 1612.95 | 5809 | 53.4 |
+| alexnet_config_2 | 0 | Enabled | 3/GPU | 44.403 | 1563.16 | 5117 | 55.1 |
+| alexnet_config_default | 0 | Disabled | 1/GPU | 8.274 | 1001.77 | 3786 | 24.6 |
+
+In the table above, we see that setting our GPU [Instance Count](https://github.com/triton-inference-server/server/blob/main/docs/model_configuration.md#instance-groups) to 5 allows us to achieve the highest throughput and lowest latency for this example. Note that our this "alexnet" model has a fixed batch-size that is explicitly specified in the first dimension of the Input/Output `dims`, therefore the `max_batch_size` parameter is set to 0 as described [here](https://github.com/triton-inference-server/server/blob/main/docs/model_configuration.md#maximum-batch-size). For models that support dynamic batch size, Model Analyzer would also tune the `max_batch_size` parameter.
```

Comment:
You should talk here about how these configs are system specific. For a smaller GPU instance_count of 5 may be counter-productive.